vasttools/query.py

```diff
@@ -1589,81 +1596,114 @@ def _get_components(self, group: pd.DataFrame) -> pd.DataFrame:
 
         return master
 
-    def _add_files(self, row: pd.Series) -> Tuple[str, str, str]:
+    def _get_selavy_path(self, epoch_string: str, row: pd.Series) -> str:
         """
-        Adds the file paths for the image, selavy catalogues and
-        rms images for each source to be queried.
-
+        Get the path to the selavy file for a specific row of the dataframe.
         Args:
-            row: The input row of the dataframe (this function is called with
-                a .apply())
-
+            epoch_string: The name of the epoch in the form of 'EPOCHXX'.
+            row: row: The input row of the dataframe.
         Returns:
-            The paths of the image, selavy catalogue and rms image.
+            The path to the selavy file of interest
         """
-        epoch_string = "EPOCH{}".format(
-            RELEASED_EPOCHS[row.epoch]
-        )
 
         if self.settings['islands']:
             cat_type = 'islands'
         else:
             cat_type = 'components'
 
         if self.settings['tiles']:
-
             dir_name = "TILES"
+            selavy_folder = Path(os.path.join(
+                self.base_folder,
+                epoch_string,
+                dir_name,
+                "STOKES{}_SELAVY_CORRECTED".format(self.settings['stokes'])
+            ))
```

Comment:
```suggestion
            selavy_folder = Path(
                self.base_folder,
                epoch_string,
                dir_name,
                f"STOKES{self.settings['stokes']}_SELAVY_CORRECTED"
            )
```

Path does the joining for you.